pennylane/capture/autograph/ag_primitives.py

```diff
@@ -0,0 +1,178 @@
+
+    results = functional_cond()
+    set_state(results)
+
+
+# Prevent autograph from converting PennyLane and Catalyst library code, this can lead to many
+# issues such as always tracing through code that should only be executed conditionally. We might
+# have to be even more restrictive in the future to prevent issues if necessary.
+module_allowlist = (
+    ag_config.DoNotConvert("pennylane"),
+    ag_config.DoNotConvert("catalyst"),
+    ag_config.DoNotConvert("optax"),
+    ag_config.DoNotConvert("jax"),
+    *ag_config.CONVERSION_RULES,
+)
+
+
+class Patcher:
+    """Patcher, a class to replace object attributes.
+
+    Args:
+        patch_data: List of triples. The first element in the triple corresponds to the object
+        whose attribute is to be replaced. The second element is the attribute name. The third
+        element is the new value assigned to the attribute.
+    """
+
+    def __init__(self, *patch_data):
+        self.backup = {}
+        self.patch_data = patch_data
+
+        assert all(len(data) == 3 for data in patch_data)
+
+    def __enter__(self):
+        for obj, attr_name, fn in self.patch_data:
+            self.backup[(obj, attr_name)] = getattr(obj, attr_name)
+            setattr(obj, attr_name, fn)
+
+    def __exit__(self, _type, _value, _traceback):
+        for obj, attr_name, _ in self.patch_data:
+            setattr(obj, attr_name, self.backup[(obj, attr_name)])
+
+
+def converted_call(fn, args, kwargs, caller_fn_scope=None, options=None):
+    """A wrapper for the autograph ``converted_call`` function, imported here as
+    ``ag_converted_call``. It returns the result of executing a possibly-converted
+     function ``fn`` with the specified ``args`` and ``kwargs``.
+
+     We want AutoGraph to use its standard behaviour with a few exceptions:
+
+       1. We want to use our own instance of the AST transformer when
+           recursively transforming functions
+       2. We want to ignore certain PennyLane modules and functions when
+           converting (i.e. don't let autograph convert them)
+       3. We want to handle QNodes, while AutoGraph generally only works on
+           functions, and to handle PennyLane wrapper functions like ctrl
+           and adjoint
+    """
+
+    # TODO: eliminate the need for patching by improving the autograph interface
+    with Patcher(
+        (ag_api, "_TRANSPILER", qml.capture.autograph.transformer.TRANSFORMER),
+        (ag_config, "CONVERSION_RULES", module_allowlist),
+    ):
+        # HOTFIX: pass through calls of known PennyLane wrapper functions
+        if fn in (
+            qml.adjoint,
+            qml.ctrl,
+            qml.grad,
+            qml.jacobian,
+            qml.vjp,
+            qml.jvp,
+        ):
+            assert args and callable(args[0])
+            wrapped_fn = args[0]
+
+            def passthrough_wrapper(*args, **kwargs):
+                return converted_call(wrapped_fn, args, kwargs, caller_fn_scope, options)
```

Comment:
```suggestion
            def passthrough_wrapper(*inner_args, **inner_kwargs):
                return converted_call(wrapped_fn, inner_args, inner_kwargs, caller_fn_scope, options)
```
Or some other sort of varation on `args` and `kwargs`, since those name are already taken in the scope.